RFP-01.md

```diff
@@ -0,0 +1,175 @@
+
+### Key Features
+
+- Ultra gas efficient signed settlement function
+- No co-mingling of funds between solvers
+- Easily observable on-chain addresses for MEV Blocker rebates
+- Permissionless access to the batch auction with reduced bonding requirements
+
+### Deliverables
+
+- Contract system for `SubPoolFactory`, `SubPool`, and `SignedSettlement`
+- Comprehensive documentation
+- Open-source code repository
+
+### Interface
+
+```solidity
+interface Auth {
+    /**
+     * Allows `usr` to perform root-level functions
+     */
+    function addOwner(address usr) external auth;
+    /**
+     * Denies `usr` from being able to perform root-level functions
+     */
+    function removeOwner(address usr) external auth;
+    /**
+     * Determine if `usr` is able to perform root-level functions
+     */
+    function isOwner(address usr) external view returns (bool);
+}
+
+interface SubPool is Auth {
+    /**
+     * Determine the amount required to `heal` the pool
+     * @return amt of immutable collateralToken
+     * @return cowAmt of COW
+     */
+    function dues() external view returns (uint256 amt, uint256 cowAmt);
+    /**
+     * Pull in from `msg.sender` the dues required to `heal` the pool
+     */
+    function heal() external;
+}
+
+interface SubPoolFactory is Auth {
+    // --- view functions ---
+
+    /**
+     * View function for determining the minimum collateral required if using `token`
+     * @param token The token to retrieve the collateral requirement for
+     */
+    function collateral(IERC20 token) external view returns (uint256);
+    /**
+     * View function for determining if a given solver EOA has a correctly collateralized
+     * sub-pool and has not announced an exit.
+     * @param subPoolSolver The solver to check
+     * @return bool if the solver is allowed to solve
+     */
+    function canSolve(address subPoolSolver) external view returns (bool);
+    /**
+     * View function for determining the sub-pool for a given solver
+     * @param solver The solver to retrieve the sub-pool for
+     * @return SubPool
+     */
+    function solverSubPool(address solver) external view returns (SubPool);
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` deterministically based on `msg.sender`.
+     * @notice The `msg.sender` SHOULD NOT be a contract.
+     * @param token The nominated token to use for collateral
+     * @param amt How much collateral to add to the sub-pool (must be minimally collateralised)
```

Comment:
What about not specifying this and just assuming that the input is `collateral(token)`?
If this is needed anyway, I suggest the following:
```suggestion
     * @param amt How much collateral to add to the sub-pool (the call will revert if
     * this value is below the minimal collateral needed for `token`)
```